test/unit/legacy-data.html

```diff
@@ -0,0 +1,278 @@
+        const el = document.createElement('x-data');
+        el._legacyUndefinedCheck = true;
+        document.body.appendChild(el);
+        assert.equal(el.static.callCount, 1);
+        assert.equal(el.one.callCount, 0);
+        assert.equal(el.two.callCount, 0);
+        assert.equal(console.warn.callCount, 0);
+        document.body.removeChild(el);
+      });
+      test('one argument defined', () => {
+        const el = document.createElement('x-data');
+        el._legacyUndefinedCheck = true;
+        el.a = 'a';
+        document.body.appendChild(el);
+        assert.equal(el.static.callCount, 1);
+        assert.equal(el.one.callCount, 1);
+        assert.equal(el.two.callCount, 0);
+        assert.equal(console.warn.callCount, 0);
+        document.body.removeChild(el);
+      });
+      test('two arguments defined', () => {
+        const el = document.createElement('x-data');
+        el._legacyUndefinedCheck = true;
+        el.a = 'a';
+        el.b = 'b';
+        document.body.appendChild(el);
+        assert.equal(el.static.callCount, 1);
+        assert.equal(el.one.callCount, 1);
+        assert.equal(el.two.callCount, 0);
+        assert.isTrue(console.warn.calledOnce);
+        assert.match(console.warn.firstCall.args[0], /'c' is undefined/);
+        document.body.removeChild(el);
+      });
+      test('all arguments defined', () => {
+        const el = document.createElement('x-data');
+        el._legacyUndefinedCheck = true;
+        el.a = 'a';
+        el.b = 'b';
+        el.c = 'c';
+        document.body.appendChild(el);
+        assert.equal(el.static.callCount, 1);
+        assert.equal(el.one.callCount, 1);
+        assert.equal(el.two.callCount, 1);
+        assert.equal(console.warn.callCount, 0);
+        document.body.removeChild(el);
+      });
+      test('one argument undefined', () => {
+        const el = document.createElement('x-data');
+        el._legacyUndefinedCheck = true;
+        el.a = 'a';
+        el.b = 'b';
+        el.c = 'c';
+        document.body.appendChild(el);
+        assert.equal(el.static.callCount, 1);
+        assert.equal(el.one.callCount, 1);
+        assert.equal(el.two.callCount, 1);
+        el.a = undefined;
+        assert.equal(el.static.callCount, 1);
+        assert.equal(el.one.callCount, 2);
+        assert.equal(el.two.callCount, 1);
+        assert.equal(console.warn.callCount, 0);
+        document.body.removeChild(el);
+      });
+      test('two arguments undefined', () => {
+        const el = document.createElement('x-data');
+        el._legacyUndefinedCheck = true;
+        el.a = 'a';
+        el.b = 'b';
+        el.c = 'c';
+        document.body.appendChild(el);
+        assert.equal(el.static.callCount, 1);
+        assert.equal(el.one.callCount, 1);
+        assert.equal(el.two.callCount, 1);
+        el.a = undefined;
+        el.b = undefined;
+        assert.equal(el.static.callCount, 1);
+        assert.equal(el.one.callCount, 2);
```

Comment:
instead of callCount spy's can we sideeffect in the calls by incrementing a counter and then the tests are totally isolated without having to spy or mock on anything.


For example

```
foo(a, b) {
   this.fooCall++;
}
```